test/controller/download.attachment.landing.controller.spec.unit.ts

```diff
@@ -0,0 +1,69 @@
+jest.mock("ioredis");
+jest.mock("../../src/middleware/authentication.middleware");
+jest.mock("../../src/middleware/session.middleware");
+jest.mock("../../src/services/objection.session.service");
+jest.mock("../../src/middleware/objection.session.middleware");
+
+import { Session } from "ch-node-session-handler/lib/session/model/Session";
+import { NextFunction, Request, Response } from "express";
+import request from "supertest";
+import app from "../../src/app";
+import { OBJECTIONS_SESSION_NAME } from "../../src/constants";
+import authenticationMiddleware from "../../src/middleware/authentication.middleware";
+import objectionSessionMiddleware from "../../src/middleware/objection.session.middleware";
+import sessionMiddleware from "../../src/middleware/session.middleware";
+import { COOKIE_NAME } from "../../src/utils/properties";
+
+const mockAuthenticationMiddleware = authenticationMiddleware as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next());
+
+const mockSessionMiddleware = sessionMiddleware as jest.Mock;
+mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+  req.session = {
+    data: {},
+  } as Session;
+  return next();
+});
+
+const mockObjectionSessionMiddleware = objectionSessionMiddleware as jest.Mock;
+mockObjectionSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+  if (req.session) {
+    req.session.data[OBJECTIONS_SESSION_NAME] = {};
+    return next();
+  }
+
+  return next(new Error("No session on request"));
+});
+
+const DOWNLOAD_LANDING_PAGE_URL: string = "/strike-off-objections/download/company/1234/strike-off-objections/5678/attachments/8888/download";
```

Comment:
the real url will start with the download prefix, i would add that to the url

Reply:
It does have the download prefix, just after the service name.

Reply:
oh yes , i do apologise